cmd/k8s-operator/svc.go

```diff
@@ -415,7 +418,7 @@ func (a *ServiceReconciler) shouldExposeDNSName(svc *corev1.Service) bool {
 }
 
 func (a *ServiceReconciler) shouldExposeClusterIP(svc *corev1.Service) bool {
-	if svc.Spec.ClusterIP == "" || svc.Spec.ClusterIP == "None" {
+	if svc.Spec.ClusterIP == "" {
```

Comment:
This changes the behaviour for Services with `None` _and_ `tailscale.com/expose` annotation- previously we would have checked if any resources for it need cleaning up, now we let it get to `maybeProvision` routine, where the validation will catch it.
The cleanup for headless `Service`s should never be needed.